api/src/main/java/org/openmrs/module/appointments/service/impl/RecurringAppointmentServiceImpl.java

```diff
@@ -135,6 +135,8 @@ private void updateMetadata(Appointment pendingAppointment, Appointment appointm
         if (appointment.getProviders() != null) {
             mapProvidersForAppointment(pendingAppointment,
                     getDeepCloneOfProviders(new ArrayList<>(appointment.getProviders())));
+        } else {
+            setRemovedProvidersToCancel(null, pendingAppointment.getProviders());
```

Comment:
These if/else conditions can be avoided by handling null case in `getDeepCloneOfProviders` method

Reply:
modified